### PR TITLE
Login validations (PR2)

### PR DESCRIPTION
**Pull request by ghost**
### Summary
* **ILoginView**
    - onEmptyEmail() added
    - onInvalidEmailFormat() added
    - onEmptyPassword() added
    - onValidatedLogin() added

* **LoginFragment**
    - SharedPreferences write / read added for login data
    -  All methods of ILoginView implemented

* **LoginPresenter**
    - onLoginValidation() method added
    - Some private methods added for modularization of the method listed above

### Card
https://trello.com/c/2AUzzjWZ/17-u1-frontend-login-validaciones

### Screen Recordings
Peek 2019-03-26 08-56

### Screenshots
image
image
Screenshot_1553264501

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java [hunk 1]
```diff
@@ -1,23 +1,94 @@
...
+import ar.com.wolox.android.example.ui.example.ExampleActivity;
 import ar.com.wolox.wolmo.core.fragment.WolmoFragment;
+import butterknife.BindView;
 
 /**
  * A simple {@link WolmoFragment} subclass.
  */
 public class LoginFragment extends WolmoFragment<LoginPresenter> implements ILoginView {
 
+    @BindView(R.id.vLoginEmailEditText)
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java]: Try to write BindViews, BindColors, BindDimens, etc. in the same line, such as the following:
@BindView(R.id.vLoginEmailEditText) EditText vLoginEmailEditText

File: app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java [hunk 2]
```diff
@@ -1,23 +1,94 @@
...
+    Button vLoginLogInButton;
+
     @Override
-    public void init() {}
+    public void init() {
+        loadLoginData();
+    }
+
+    private void loadLoginData() {
+        Context context = getActivity();
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java]: There is no need to use this line. You can use directly getContext() method

File: app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java [hunk 3]
```diff
@@ -1,23 +1,94 @@
...
+    @Override
+    public void onValidatedLogin() {
+        saveLoginDataOnSharedPreferences(vLoginEmailEditText.getText().toString(),
+                vLoginPasswordEditText.getText().toString());
+        Intent intent = new Intent(getContext(), ExampleActivity.class);
+        startActivity(intent);
+    }
+
+    private void saveLoginDataOnSharedPreferences(String email, String password) {
+        Context context = getActivity();
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java]: Same as above

File: app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginPresenter.java [hunk 4]
```diff
@@ -11,4 +12,39 @@
...
+        if (isValidEmail && isValidPassword) {
+            getView().onValidatedLogin();
+        }
+    }
+
+    private boolean validateEmail(String email) {
+        boolean isValidEmail = false;
+
+        if (TextUtils.isEmpty(email)) {
+            getView().onEmptyEmail();
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginPresenter.java:29]: I would prefer that in this line appears 'isValidEmail = false' and then in onLoginValidation you decide where to go

File: app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginPresenter.java [hunk 5]
```diff
@@ -11,4 +12,39 @@
...
+        }
+
+        return isValidEmail;
+    }
+
+    private boolean validatePassword(String password) {
+        boolean isValidPassword = false;
+
+        if (TextUtils.isEmpty(password)) {
+            getView().onEmptyPassword();
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginPresenter.java:43]: Same as above

**Review by lmiotti — CHANGES_REQUESTED** [review]
In general it's quite good. We have to do some minor changes, but in general is in condition to be approved. 
P.D: We can add a GIF to the PR :smile:

File: app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java (hunk 1 above)
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java]: ✔ **Done**

File: app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java (hunk 3 above)
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java]: ✔ **Done**

File: app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java (hunk 2 above)
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginFragment.java]: ✔ **Done**
Interesting correction

**Comment by ghost** [comment]
> P.D: We can add a GIF to the PR :smile: 

GIF added! :sunglasses:

**Review by lmiotti — APPROVED** [review]
superheroes

File: app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginPresenter.java (hunk 4 above)
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginPresenter.java:29]: ✔ **Discussed in morning call:** We agreed to keep it in the way it is

File: app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginPresenter.java (hunk 5 above)
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginPresenter.java:43]: ✔ **Same resolution as above**

**Pull request merged**